File: examples/Ruby/rrclient.rb
# author: Oleg Sidorov <4pcbr> i4pcbr@gmail.com
# this code is licenced under the MIT/X11 licence.

require 'rubygems'
require 'ffi-rzmq'

context = ZMQ::Context.new
socket = context.socket(ZMQ::REQ)
socket.connect('tcp://localhost:5559')

10.times do |request|
  string = "Hello #{request}"
  socket.send_string(string)
  puts "Sending string [#{string}]"
  socket.recv_string(message = '')
  puts "Received reply #{request}[#{message}]"
end

File: examples/Rust/rrclient.rs
#![crate_name = "rrclient"]

extern crate zmq;

fn main() {
    let context = zmq::Context::new();

    let requester = context.socket(zmq::REQ).unwrap();
    assert!(requester.connect("tcp://localhost:5559").is_ok());

    for request_nbr in 0..10 {
        requester.send_str("Hello", 0).unwrap();
        let string = requester.recv_string(0).unwrap().unwrap();
        println!("Received reply {} {}", request_nbr, string);
    }
}
